packages/wrangler/templates/startDevWorker/InspectorProxyWorker.ts

```diff
@@ -0,0 +1,472 @@
+// TODO: Inspector Durable Object
+//  1. ~~Durable Object websocket server~~
+//  2. ~~JSON list endpoints (proxy through to workerd and filter for core:user:*)~~
+//  3. ~~Receive cf data from proxy controller (use env.SECRET for auth)~~
+//  4. ~~Connect to remote websocket and establish devtools connection~~
+//  5. ~~Buffer messages from remote until we've got client connection~~
+//  6. ~~Intercept logging messages and log them to the console (probably via websocket to proxy controller)~~
+//  7. Rewriting messages for source maps and stuff
+//
+
+/**
+ * TODO:
+ *
+ * - Rewriting messages for source maps and stuff
+ *      -
+ * - tests
+ *      - test devtools get-source-map triggers service binding callback?
+ *      -
+ */
+
+import assert from "node:assert";
+import {
+	castErrorCause,
+	DevToolsCommandRequest,
+	DevToolsCommandRequests,
+	DevToolsCommandResponse,
+	DevToolsCommandResponses,
+	DevToolsEvent,
+	DevToolsEvents,
+	type InspectorProxyWorkerIncomingWebSocketMessage,
+	type InspectorProxyWorkerOutgoingRequestBody,
+	type InspectorProxyWorkerOutgoingWebsocketMessage,
+	type ProxyData,
+} from "../../src/api/startDevWorker/events";
+import {
+	MaybePromise,
+	assertNever,
+	createDeferredPromise,
+} from "../../src/api/startDevWorker/utils";
+
+interface Env {
+	PROXY_CONTROLLER: Fetcher;
+	PROXY_CONTROLLER_AUTH_SECRET: string;
+	WRANGLER_VERSION: string;
+	DURABLE_OBJECT: DurableObjectNamespace;
+}
+
+export default {
+	fetch(req, env) {
+		const singleton = env.DURABLE_OBJECT.idFromName("");
+		const inspectorProxy = env.DURABLE_OBJECT.get(singleton);
+
+		return inspectorProxy.fetch(req);
+	},
+} as ExportedHandler<Env>;
+
+export class InspectorProxyWorker implements DurableObject {
```

Comment:
This is a big boi, sorry

It's job is to proxy messages between the runtime inspector (`#websockets.runtime`), ProxyController (`#websockets.proxyController`) and devtools client (#websockets.devtools).

Because only one client can be connected to the runtime inspector server, the InspectorProxyWorker is essentially that one client. The ProxyController and devtools then can both connect to the InspectorProxyWorker.